### PR TITLE
Make demo directly install citadel

### PR DESCRIPTION
**Pull request by howardjohn**
Currently the demo will generate config for all components but then
reference this pregenerated config for citadel. This changes it to align
with other components and be self contained.

Ran make run-build-demo which is where most changes here come from

### CODE REVIEW

**Comment by howardjohn** [comment]
The hold stuff was just testing github - if someone actually approves this feel free to remove it

File: test/demo/k8s.yaml [hunk 1]
```diff
@@ -12,7 +12,7 @@ metadata:
     istio: sidecar-injector
```
Comment (costinm) [inline test/demo/k8s.yaml]: This should probably move under 'kustomize/' directory, for consistency ( it was created before we started to pre-generate profiles ). Can be separate PR.

File: test/demo/values.yaml [hunk 2]
```diff
@@ -3,6 +3,16 @@
 # Includes components used in the demo, defaults to alpha3 rules.
 # Note: please only put common configuration for the demo profiles here.
 global:
+  istioNamespace: istio-system
```
Comment (costinm) [inline test/demo/values.yaml]: It would be really great if the demo used the split-namespace model. 
It's a demo, intended for fresh install/remove in non-prod clusters (since it takes over istio-system ) - no upgrade concerns.

File: test/demo/k8s.yaml (hunk 1 above)
Comment (howardjohn) [reply test/demo/k8s.yaml]: We could do this, but it would require to redefine the demo profile from being a set of values.yaml to kustomizations, right?

File: test/demo/values.yaml (hunk 2 above)
Comment (howardjohn) [reply test/demo/values.yaml]: Is this still the plan with the new same-namespace upgrade plan you made? Or would that be each component upgrades in the same namespace, but we still have split namespaces?

File: test/demo/k8s.yaml (hunk 1 above)
Comment (costinm) [reply test/demo/k8s.yaml]: Not really. We can provide one 'kustomization' for each profile - or just support a single profile, do we really need all the demo variants ?

File: test/demo/values.yaml (hunk 2 above)
Comment (costinm) [reply test/demo/values.yaml]: Let's go with the simplest - all in istio-system.
